src/python/test/test_extract_expression.py

```diff
@@ -78,8 +116,400 @@ def standard_positive_filter_test(self, json_name):
+        self.standard_input_args_test(input_dict, expected_error_message)
+
+    # EM-16
+    # When the string provided is a malformed JSON
+    def test_malformed_retr_exp_json(self):
+        expected_error_message = (
+            "JSON provided for --retrieve-expression is malformatted."
+        )
+        command = [
+            "dx",
+            "extract_assay",
+            "expression",
+            self.test_record,
+            "--retrieve-expression",
+            r"{thisisbadjson",
+        ]
+
+        process = subprocess.Popen(
+            command, stderr=subprocess.PIPE, universal_newlines=True
+        )
+        actual_err_msg = process.communicate()[1]
+        # print(actual_err_msg)
+
+        self.assertTrue(expected_error_message in actual_err_msg)
+
+    # EM-17
+    # When the .json file provided does not exist
+    # Note: this probably needs to be tested with a Popen rather than with the ValidateArgsBySchema function
+    def test_json_file_not_exist(self):
+        missing_json_path = os.path.join(self.general_input_dir, "nonexistent.json")
+        expected_error_message = (
+            "JSON file provided to --retrieve-expression does not exist".format(
+                missing_json_path
+            )
+        )
+        command = [
+            "dx",
+            "extract_assay",
+            "expression",
+            self.test_record,
+            "--retrieve-expression",
+            missing_json_path,
+        ]
+
+        process = subprocess.Popen(
+            command, stderr=subprocess.PIPE, universal_newlines=True
+        )
+        actual_err_msg = process.communicate()[1]
+        # print(actual_err_msg)
+
+        self.assertTrue(expected_error_message in actual_err_msg)
+
+    # EM-21
+    # When --json-help is passed with another option from --assay-name, --sql, --additional-fields, --expression-matix, --output
+    def test_json_help_other_option(self):
+        expected_error_message = "--json-help cannot be passed with any of --assay-name, --sql, --additional-fields, --expression-matix, or --output"
+        input_dict = {
+            "path": self.test_record,
+            "json_help": True,
+            "assay_name": "test_assay",
+        }
+        self.standard_input_args_test(input_dict, expected_error_message)
+
+    # EM-22
+    # When --expression-matrix is passed with other arguments other than, any context other than, --retrieve-expression
+    # It seems that every combination of args that could be passed with this cause a different issue to be caught first
+    # Which is fine but the error message will be for the other error
+    def test_exp_matrix_other_args(self):
+        # expected_error_message = "--expression-matrix cannot be passed with any argument other than --retrieve-expression"
+        expected_error_message = "“--json-help cannot be passed with any of --assay-name, --sql, --additional-fields, --expression-matix, or --output”"
```

Comment:
It's probably fine to just pull the expected message directly from the schema too. @jkotrs any preferences there? Should we hardcode the expected error message or is it fine to grab it from the schema?

```EXTRACT_ASSAY_EXPRESSION_INPUT_ARGS_SCHEMA.get("5_json_help_exclusive_with_exceptions").get("message)```

Reply:
If possible it would be even better to grab it from the spec.  If we grab the error message from the schema, then we won't catch the case where the correct error is being caught, but the message in the schema is wrong